library/src/main/java/com/jwoolston/android/uvc/StreamManager.java

```diff
@@ -45,26 +51,102 @@
-        Timber.d("Request Result: %s", (retval > 0 ? retval : LibusbError.fromNative(retval)));
-        Timber.d("Response data: 0x%s\n", Hexdump.dumpHexString(current.getData()));
+        int maxPayload = current.getMaxPayloadTransferSize();
+        int maxFrameSize = current.getMaxVideoFrameSize();
 
-        final ProbeControl request = ProbeControl.setCurrentProbe(streamingInterface);
-        request.setFormatIndex(format != null ? format.getFormatIndex() : 1);
-        request.setFrameIndex(frame != null ? frame.getFrameIndex() : 1);
+        final ProbeControl commit = current.getCommit();
+
+        retval = connection.controlTransfer(commit.getRequestType(), commit.getRequest(),
+                                            commit.getValue(), commit.getIndex(), commit.getData(), commit.getLength(),
+                                            500);
+        if (retval < 0) {
+            throw new StreamCreationException("Commit request failed: " + LibusbError.fromNative(retval));
+        }
+
+        final RequestErrorCode requestErrorCode = RequestErrorCode.getCurrentErrorCode(controlInterface);
+        retval = connection.controlTransfer(requestErrorCode.getRequestType(), requestErrorCode.getRequest(),
+                                            requestErrorCode.getValue(), requestErrorCode.getIndex(),
+                                            requestErrorCode.getData(), requestErrorCode.getLength(), 500);
+        if (retval < 0 || requestErrorCode.getData()[0] != 0) {
+            throw new StreamCreationException("Error state failed: " + (retval < 0 ? LibusbError.fromNative(retval)
+            : "Current error code: 0x" + Hexdump.toHexString(requestErrorCode.getData()[0])));
+        }
+
+        Timber.d("Current error code: 0x%s", Hexdump.toHexString(requestErrorCode.getData()[0]));
+
+        initiateStream(maxPayload, maxFrameSize);
+    }
+
+    public void initiateStream(int maxPayload, int maxFrameSize) {
+        streamingInterface.selectAlternateSetting(connection, 6);
+        ByteBuffer buffer = ByteBuffer.allocateDirect(maxPayload);
+        Endpoint endpoint = streamingInterface.getCurrentEndpoints()[0];
+        try {
+            IsochronousAsyncTransfer transfer = new IsochronousAsyncTransfer(this, endpoint.getEndpoint(),
+                                                                             connection, 20);
+            transfer.submit(buffer, 500);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public void onIsochronousTransferComplete(@Nullable ByteBuffer data, int result) throws IOException {
+        if (result < 0) {
+            throw new IOException("Failure in isochronous callback:" + LibusbError.fromNative(result));
+        } else {
+            final byte[] raw = new byte[data.limit()];
+            data.rewind();
+            data.get(raw);
+            Timber.d(" \n%s", Hexdump.dumpHexString(raw));
```

Comment:
I assume this is temporary otherwise the hexdump needs to be conditionally run such as only for debug builds or the like.

Reply:
Yes its just until the data is actually going somewhere.